src/app/website/posts/zkp.markdown

```diff
@@ -0,0 +1,81 @@
+---
+title: Zero-knowledge Proofs&#58; An Intuitive Explanation
+date: 2019-05-13
+author: Vanishree Rao
+---
+
+Zero-knowledge proofs (ZKPs) are a powerful cryptography primitive that enables you to prove that you have a secret, without revealing it to anyone.
+ If you are hearing about ZKPs for the first time, you are likely to say "Hah! That sounds impossible." Read on to get an intuitive understanding of what they are. But first, some background. ZKPs were [invented](http://people.csail.mit.edu/silvio/Selected%20Scientific%20Papers/Proof%20Systems/The_Knowledge_Complexity_Of_Interactive_Proof_Systems.pdf) by Shafi Goldwasser, Silvio Micali, and Charles Rackoff in 1985. Ever since, ZKPs have been one of the most active areas of research in Cryptography. Moreover, recently, they are enjoying significant impact on real-world applications, specifically on blockchain technologies. [Zcash](https://z.cash/technology/), a pioneering blockchain project, employed ZKPs to achieve anonymity in financial transactions. At O(1)Labs, we are building CODA, the first succinct blockchain, using ZKPs. No matter how many transactions are recorded on the blockchain, the blockchain remains at most the size of a few tweets. 
+
+
+
+## What are Zero-knowledge Proofs?
+
+Now let’s consider a puzzle and see how you could convince anybody that you know a solution without revealing it. 
```

Comment:
I think this could use a segue. Maybe something like

The purpose of zero-knowledge proofs is to convince someone you know *something* without revealing what that thing is. For example, you might want to convince someone that you know the solution to a puzzle without giving them the solution.

Let's look at a particular puzzle to see how you can accomplish exactly this.

Reply:
@vanishreerao  for me, there where's waldo with an oversized mask is the canonical example -- not sure where I saw that.

Reply:
Done. @jkrauska: nice, I found it: https://medium.com/swlh/a-zero-knowledge-proof-for-wheres-wally-930c21e55399
nice read!